module/batch-action/src/Application/Transport/BatchActionTransport.php

```diff
@@ -16,52 +16,63 @@
 use Ergonode\SharedKernel\Domain\AggregateId;
 use Symfony\Component\Messenger\Stamp\HandledStamp;
 use Ergonode\BatchAction\Domain\Repository\BatchActionRepositoryInterface;
+use Ergonode\Core\Application\Security\User\CachedUser;
+use Ergonode\Account\Domain\Repository\UserRepositoryInterface;
+use Ergonode\SharedKernel\Domain\Aggregate\UserId;
+use Ergonode\BatchAction\Domain\Event\BatchActionEndedEvent;
+use Ergonode\Core\Application\Messenger\Stamp\UserStamp;
 
 class BatchActionTransport implements TransportInterface
 {
     private Connection $connection;
 
     private BatchActionRepositoryInterface $repository;
 
-    public function __construct(Connection $connection, BatchActionRepositoryInterface $repository)
-    {
+    private UserRepositoryInterface $userRepository;
+
+    public function __construct(
+        Connection $connection,
+        BatchActionRepositoryInterface $repository,
+        UserRepositoryInterface $userRepository
+    ) {
         $this->connection = $connection;
         $this->repository = $repository;
+        $this->userRepository = $userRepository;
     }
 
     public function get(): iterable
     {
         $this->connection->beginTransaction();
-        $result = [];
-        $record = $this->connection->executeQuery(
-            'SELECT batch_action_id, resource_id 
-                 FROM batch_action_entry 
-                 WHERE processed_at is NULL LIMIT 1 FOR UPDATE SKIP LOCKED'
-        )->fetchAssociative();
 
-        if (!empty($record)) {
-            echo ($record['resource_id']).PHP_EOL;
-            $result[] = new Envelope(
-                new ProcessBatchActionEntryCommand(
-                    new BatchActionId($record['batch_action_id']),
-                    new AggregateId($record['resource_id'])
-                )
-            );
-        } else {
-            $this->connection->commit();
+        $envelope = $this->getEntryMessage();
+        if (null === $envelope) {
+            $envelope = $this->getBatchActionMessage();
+        }
+
+        if ($envelope) {
+            return [$envelope];
         }
 
-        return $result;
+        $this->connection->commit();
```

Comment:
As far as I can see you're not rollbacking transaction anywhere. This could potentially cause further issues(and very hard to debug ones) with not synchronized transactions between message handling.
Is the transaction required in this place anyway? You seem only to be fetching data here. ack method is performing some updates so it makes sense